### PR TITLE
Clean up some of the extensions

### PR DESCRIPTION
**Pull request by anthonydresser**
Adds more compile options to some of the extensions.

Removes unused code.

Remove some unnecessary node modules

### CODE REVIEW

File: extensions/big-data-cluster/src/commands.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export const ManageControllerCommand = 'bigDataClusters.command.manageController';
```
Comment (anthonydresser) [inline extensions/big-data-cluster/src/commands.ts]: There was a circular dependency with these consts; so I moved them into their own file.

File: src/sql/workbench/api/common/sqlExtHostTypes.ts [hunk 2]
```diff
@@ -697,9 +697,9 @@ export class ConnectionProfile {
 		this.options['azureTenantId'] = value;
 	}
 
-	options: Map<string, any> = new Map<string, any>();
+	options: { [key: string]: any } = {};
 
-	static createFrom(options: Map<string, any>): ConnectionProfile {
+	static createFrom(options: { [key: string]: any }): ConnectionProfile {
```
Comment (anthonydresser) [inline src/sql/workbench/api/common/sqlExtHostTypes.ts]: Found a bug here. Map =/= {}. how this is currently being used it is a json object not a map.

File: extensions/mssql/src/objectExplorerNodeProvider/hdfsCommands.ts [hunk 3]
```diff
@@ -71,8 +71,8 @@ export class UploadFilesCommand extends ProgressCommand {
 		try {
 			let folderNode = await getNode<FolderNode>(context, this.appContext);
 			const allFilesFilter = localize('allFiles', "All Files");
-			let filter = {};
-			filter[allFilesFilter] = '*';
+			let filter: { [key: string]: string[] } = {};
+			filter[allFilesFilter] = ['*'];
```
Comment (anthonydresser) [inline extensions/mssql/src/objectExplorerNodeProvider/hdfsCommands.ts]: Functional change here. This is defined by the vscode apis. I suspect the only reason it was working before is vscode had some back compat code in place to catch this.

File: extensions/mssql/src/main.ts [hunk 4]
```diff
@@ -181,8 +181,8 @@ async function handleNewNotebookTask(oeContext?: azdata.ObjectExplorerContext, p
 
 async function handleOpenNotebookTask(profile: azdata.IConnectionProfile): Promise<void> {
 	let notebookFileTypeName = localize('notebookFileType', "Notebooks");
-	let filter = {};
-	filter[notebookFileTypeName] = 'ipynb';
+	let filter: { [key: string]: string[] } = {};
+	filter[notebookFileTypeName] = ['ipynb'];
```
Comment (anthonydresser) [inline extensions/mssql/src/main.ts]: As defined by the vscode api

File: extensions/mssql/src/hdfs/webhdfs.ts [hunk 5]
```diff
@@ -794,7 +795,7 @@ export class WebHDFS {
 			canResume = false;
 			stream.on('resume', () => {
 				if (!canResume) {
-					stream._readableState.flowing = false;
+					(stream as any)._readableState.flowing = false; // i guess we are unsafely accessing this
```
Comment (Charles-Gagnon) [inline extensions/mssql/src/hdfs/webhdfs.ts]: @kevcunnane - Thoughts? We should be safer here and add a check for _readableState being defined. And if it isn't then  probably emit an error. I imagine never an actual issue but better to be safe

Comment (kevcunnane) [reply extensions/mssql/src/hdfs/webhdfs.ts]: this is a workaround we inherited from the public WebHDFS module...  I think this is fine and overall an improvement. Happy to have a comment mentioning it'd be nice not to have to reference this.

**Pull request merged**